docs/csharp/language-reference/operators/null-forgiving.md

```diff
@@ -13,6 +13,9 @@ Available in C# 8.0 and later, the unary postfix `!` operator is the null-forgiv
 
 The null-forgiving operator has no effect at run time. It only affects the compiler's static flow analysis by changing the null state of the expression. At run time, expression `x!` evaluates to the result of the underlying expression `x`.
 
+> [!NOTE]
+> The null-forgiving operator can interact with the [null-conditional operators](member-access-operators.md#null-conditional-operators--and-) in unexpected ways. In C# 8 the expression `x?.y!.z` is parsed as `(x?.y)!.z`. Due to this interpretation `z` is evaluated even if `x` is null, which may result in a <xref:System.NullReferenceException>.
```

Comment:
```suggestion
> In C# 8.0, a [null-conditional](member-access-operators.md#null-conditional-operators--and-) expression doesn't include the null-forgiving operator. That is, expression `x?.y!.z` is parsed as `(x?.y)!.z`. Due to this interpretation, `z` is evaluated even if `x` is `null`, which may result in a <xref:System.NullReferenceException>.
```

Reply:
That's a good point @pkulikov I made some edits, but I did leave the word "unexpected" in. In this instance, it's appropriate.

Reply:
@BillWagner thanks, then I have two nit-picks:
- Should "can interact" be "interacts"? As "can interact" suggests that it doesn't interact always like that
- Should "unexpected ways" be "unexpected way"?